sound/soc/sof/sof-priv.h

```diff
@@ -93,6 +93,9 @@ struct snd_sof_dsp_ops {
 	int (*probe)(struct snd_sof_dev *sof_dev); /* mandatory */
 	int (*remove)(struct snd_sof_dev *sof_dev); /* optional */
 
+	/* machine driver probing */
+	int (*machine_register)(struct snd_sof_dev *sof_dev); /* mandatory */
+
```

Comment:
should this be optional?

Reply:
I think yes, although I haven't seen any platform which doesn't have a "codec" scenario.